tests/unit/processing_container/test_process.py

```diff
@@ -93,3 +103,87 @@ def test_main(self, mock_rematser_cxg_process, mock_download_from_source_uri):
+
+    @patch("backend.corpora.dataset_processing.process_cxg.process")
+    @patch("backend.corpora.dataset_processing.process_seurat.process")
+    @patch("backend.corpora.dataset_processing.process_download_validate.process")
+    def test_main__Exceptions(self, mock_process_download_validate, mock_process_seurat, mock_process_cxg):
+        dataset = self.generate_dataset(self.session, collection_id="test_collection_id")
+        test_environment = {
+            "DROPBOX_URL": "https://www.dropbox.com/IGNORED",
+            "ARTIFACT_BUCKET": self.corpora_config.bucket_name,
+            "CELLXGENE_BUCKET": self.corpora_config.bucket_name,
+            "DATASET_ID": dataset.id,
+            "DEPLOYMENT_STAGE": "test",
+        }
+        with self.subTest("download-validate unknown exceptions"):
+            mock_process_download_validate.side_effect = Exception(
+                {"validation_status": ValidationStatus.INVALID, "validation_message": "Anndata could not open raw.h5ad"}
+            )
+            test_environment["STEP_NAME"] = "download-validate"
+            with EnvironmentSetup(test_environment):
+                main()
+                expected_dataset = Dataset.get(self.session, test_environment["DATASET_ID"]).processing_status
+                self.assertEqual(expected_dataset.validation_status, ValidationStatus.INVALID)
+                self.assertEqual(expected_dataset.validation_message, "Anndata could not open raw.h5ad")
+
+            mock_process_download_validate.side_effect = Exception("unknown failure")
+            dataset = self.generate_dataset(self.session, collection_id="test_collection_id")
+            test_environment["DATASET_ID"] = dataset.id
+            with EnvironmentSetup(test_environment):
+                main()
+                expected_dataset = Dataset.get(self.session, test_environment["DATASET_ID"]).processing_status
+                self.assertEqual(expected_dataset.upload_status, UploadStatus.FAILED)
+                self.assertEqual(expected_dataset.upload_message, "unknown failure")
+
+        with self.subTest("download-validate known exceptions"):
+            mock_process_download_validate.side_effect = ValidationFailed(
+                {"validation_status": ValidationStatus.INVALID, "validation_message": "Schema version is not supported"}
+            )
+            dataset = self.generate_dataset(self.session, collection_id="test_collection_id")
+            test_environment["DATASET_ID"] = dataset.id
+            with EnvironmentSetup(test_environment):
+                main()
+                expected_dataset = Dataset.get(self.session, test_environment["DATASET_ID"]).processing_status
+                self.assertEqual(expected_dataset.validation_status, ValidationStatus.INVALID)
+                self.assertEqual(expected_dataset.validation_message, "Schema version is not supported")
+
+            mock_process_download_validate.side_effect = ProcessingFailed(
```

Comment:
L151-160: should be its own subtest. You can use the side_effect value as the name.